components/server/src/workspace/workspace-starter.ts

```diff
@@ -607,8 +607,12 @@ export class WorkspaceStarter {
             user.additionalData.ideSettings = migratted;
         }
 
-        const ideChoice = user.additionalData?.ideSettings?.defaultIde;
-        const useLatest = !!user.additionalData?.ideSettings?.useLatestVersion;
+        const preferenceIDE = user.additionalData?.ideSettings?.defaultIde;
+        const perferenceUseLatest = !!user.additionalData?.ideSettings?.useLatestVersion;
+        const ideChoice = WithEditorContext.is(workspace.context) ? workspace.context.ide : preferenceIDE;
+        const useLatest = WithEditorContext.is(workspace.context)
+            ? workspace.context.useLatest ?? false
```

Comment:
actually maybe it makes sense that version falls back to user preferences if not provided?